shared/reports/resources.py

```diff
@@ -587,6 +587,17 @@ def line_without_multiple_sessions(
         )
 
 
+def chunks_from_storage_contains_header(chunks: str) -> bool:
```

Comment:
This looks good as long as the header JSON never contains newlines

Reply:
hum... fair point. Labels can get pretty wild. I think we should assume that the labels in the header might contain newlines.

Do you have any suggestions for the function?

Reply:
So I did test that and it seems we're fine even if the labels have "\n" in them cause they'll be JSON stringified... so I guess it's fine (at least for now) to leave as is.

```python
>>> d = {"strange_label": "big label with \n \n in it"}
>>> print(d["strange_label"])
big label with 
 
 in it
>>> x = json.dumps(d)
>>> x
'{"strange_label": "big label with \\n \\n in it"}'
>>> "\n" in x
False
>>> x.index("\n")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: substring not found
>>> x.index('\n')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: substring not found
>>> x.split("\n")
['{"strange_label": "big label with \\n \\n in it"}']

Reply:
The alternative

```python
def chunks_from_storage_contains_header(chunks: str) -> bool:
    possible_first_chunk, _ = chunks.split(END_OF_CHUNK, 1)
    return END_OF_HEADER in possible_first_chunk
```

is most likely less fast that the current implementation. Cause the `possible_first_chunk` might be pretty big.